Subscription pallet (#147)
dappforce/subsocial-parachain: pallets/subscriptions/README.md

> # Subscriptions Pallet

Reviewer: @Yung-Beef plz check

> Subscribing to a space is something similar to following a space, except subscribers will be granted a role for the space. This role can have any set of permissions space owner chooses to allow.

Reviewer 2: ```suggestion
Subscribing to a space is similar to following a space, except subscribers will be granted a role in the space. This role can have any set of permissions that the space owner chooses to allow.
```

For the first version of this pallet subscription is meant to be only one time payment. But in the future we will allow recurring subscription.

The space owner at any time can disable this role, and sets a new subscription settings with a new role.

The space owner can change the subscription value and it will be effective for new subscribers. 

> Old subscribers can still enjoy the role they have been granted as long as the role is not disabled.

Reviewer 2: ```suggestion
Old subscribers can still enjoy their roles as long as those roles have not disabled.
```
